keystrokes/src/components/Character/CharacterDetails/KeybindEditor.js

```diff
@@ -0,0 +1,191 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+	Grid,
+	Button,
+	Typography,
+	Menu,
+	MenuItem,
+	CircularProgress,
+	ExpansionPanel,
+	ExpansionPanelSummary,
+	ExpansionPanelDetails,
+} from '@material-ui/core';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import styleGuide from '../../../stylesheets/style';
+import { ref } from '../../../config/constants';
+import { characterKeybindings } from '../../utils/utils';
+import ManualKeybindModal from './ManualKeybindModal';
+import KeybindTable from './KeybindTable';
+import RapidFireKeybindModal from './RapidFireKeybindModal';
+
+KeybindEditor.propTypes = {
+	character: PropTypes.object,
+	setCharacter: PropTypes.func,
+	spec: PropTypes.number,
+	allKeybindings: PropTypes.object,
+	setAllKeybindings: PropTypes.func,
+	keyBinding: PropTypes.number,
+};
+function KeybindEditor({
+	character,
+	setCharacter,
+	spec,
+	allKeybindings,
+	setAllKeybindings,
+	keyBinding,
+}) {
+	const classes = styleGuide();
+	const [editOptions, setEditOptions] = useState();
+	const [rapidFireModal, setRapidFireModal] = useState(false);
+	const [manualModal, setManualModal] = useState(false);
+	const [loading, setLoading] = useState(true);
+
+	useEffect(() => {
+		async function collectKeybindings() {
+			const path = `/Keybindings/${characterKeybindings(
+				character,
+				spec,
+				keyBinding
+			)}`;
+			const snapShot = await ref.child(path).once('value');
+			if (
+				snapShot.exists() &&
+				!Object.keys(allKeybindings).includes(
+					characterKeybindings(character, spec, keyBinding)
+				)
+			) {
+				const keybindingsWeGot = snapShot.val();
+				setAllKeybindings({
+					...allKeybindings,
+					[characterKeybindings(
+						character,
+						spec,
+						keyBinding
+					)]: keybindingsWeGot,
+				});
+			}
+			//Set something to say "character not found"
+			setLoading(false);
+		}
+		collectKeybindings();
+	});
+
+	const handleClick = event => {
+		setEditOptions(event.currentTarget);
+	};
+
+	const handleClose = () => {
+		setEditOptions();
+	};
+	return (
+		<React.Fragment>
```

Comment:
fyi there's a short had to fragment that's just <> </>

Reply:
Ive actually had issues with that in the past, sometimes it doesnt work for both syntaxes. ill try to see if i can find an example of it not working, otherwise ill start using the <></>